muonDIS/makeMuonDIS.py

```diff
@@ -0,0 +1,204 @@
+        muPart = r.TVectorD(12, mu)
+        myPythia.Initialize('FIXT', mutype[pid], 'p+', p)
+        myPythia.Pylist(1)
+        target='p+'
+        
+        for a in range(args.nDIS):
+            if a == args.nDIS // 2:
+                myPythia.Initialize('FIXT', mutype[pid], 'n0', p)
+                isProton = 0
+                target='n0'
+                print(f"Switching to neutron interaction")
+
+            dPartDIS.Clear()
+            iMuon.Clear()
+            muPart[9] = isProton
+            iMuon[0] = muPart
+            myPythia.GenerateEvent()
+            myPythia.Pyedit(1)
+            print(f"Event {a} generated, number of particles: {myPythia.GetN()}")
+
+            
+            for itrk in range(1, myPythia.GetN() + 1):
+                xsec = myPythia.GetPARI(1)
+                muPart[10] = xsec
+                did = myPythia.GetK(itrk, 2)
+                dpx, dpy, dpz = rotate(ctheta, stheta, cphi, sphi, myPythia.GetP(itrk, 1), myPythia.GetP(itrk, 2), myPythia.GetP(itrk, 3))
+                psq = dpx ** 2 + dpy ** 2 + dpz ** 2
+                E = r.TMath.Sqrt(getMasssq(did) + psq)
+                m = array('d', [did, dpx, dpy, dpz, E])
+                part = r.TVectorD(5, m)
+                nPart = dPartDIS.GetEntries()
+                if dPartDIS.GetSize() == nPart:
+                    dPartDIS.Expand(nPart + 10)
+                dPartDIS[nPart] = part
+                if itrk == 1:
+                    with open(f'sigmadata_{args.nJob}.txt', "a") as fcross:
+                        fcross.write(f"{xsec}\n")
+            
+            
+            dPartSoft.Clear()
+            
+            softTracks = sTree.tracks
+
+            for itrk in range(softTracks.GetEntries()):
+                
+                softTrack = softTracks[itrk]
+                            
+                
+                did = softTrack.GetPdgCode()
+                dpx = softTrack.GetPx()
+                dpy = softTrack.GetPy()
+                dpz = softTrack.GetPz()
+                
+                psq = dpx ** 2 + dpy ** 2 + dpz ** 2
+                E = r.TMath.Sqrt(getMasssq(did) + psq)
+                
+                softx = softTrack.GetStartX()
+                softy = softTrack.GetStartY()
+                softz = softTrack.GetStartZ()
+                time_ = softTrack.GetStartT()
+                
+                m = array('d', [did, dpx, dpy, dpz, E,softx,softy,softz,time_])
+                
+                part = r.TVectorD(9, m)
+                nPart = dPartSoft.GetEntries()
+                if dPartSoft.GetSize() == nPart:
+                    dPartSoft.Expand(nPart + 10)
+                dPartSoft[nPart] = part
+            
+            muon_vetoPoints.Clear()
+            index=0
+
+            for hit in sTree.muon_vetoPoints:
+                if muon_vetoPoints.GetSize() == index:
+                    muon_vetoPoints.Expand(index+1)
+                muon_vetoPoints[index] = hit
```

Comment:
This should be adapted for ROOT 6.32 as done in https://github.com/ShipSoft/FairShip/pull/559